src/api_client/api_client.js

```diff
@@ -69,29 +59,36 @@ export default class APIClient {
 	}
 
 	static createMainnetAPIClient(options) {
-		return new APIClient(MAINNET_NODES, MAINNET_NETHASH, options);
+		return new APIClient(
+			MAINNET_NODES,
+			Object.assign({}, options, { nethash: MAINNET_NETHASH }),
+		);
 	}
 
 	static createTestnetAPIClient(options) {
-		return new APIClient(TESTNET_NODES, TESTNET_NETHASH, options);
+		return new APIClient(
+			TESTNET_NODES,
+			Object.assign({}, options, { nethash: TESTNET_NETHASH }),
+		);
 	}
 
 	static createBetanetAPIClient(options) {
-		return new APIClient(BETANET_NODES, BETANET_NETHASH, options);
+		return new APIClient(
+			BETANET_NODES,
+			Object.assign({}, options, { nethash: BETANET_NETHASH }),
+		);
 	}
 
-	initialize(nodes, nethash, providedOptions = {}) {
+	initialize(nodes, providedOptions = {}) {
 		if (!Array.isArray(nodes) || nodes.length <= 0) {
 			throw new Error('APIClient requires nodes for initialization.');
 		}
 
-		if (typeof nethash !== 'string' || nethash === '') {
-			throw new Error('APIClient requires nethash for initialization.');
-		}
-
 		const options = Object.assign({}, defaultOptions, providedOptions);
 
-		this.headers = getHeaders(nethash, options.version, options.minVersion);
+		this.headers = options.nethash
+			? Object.assign({}, commonHeaders, { nethash: options.nethash })
+			: commonHeaders;
```

Comment:
We should have a fresh object here in case people override `myClientInstance.headers` and don't expect it to affect other clients.